Encoder/DefaultEncoder.php

```diff
@@ -29,10 +29,10 @@ public function __construct(JWSProviderInterface $jwsProvider)
     /**
      * {@inheritdoc}
      */
-    public function encode(array $payload)
+    public function encode(array $header, array $payload)
```

Comment:
Btw, let's make $header the last arg, that will give a simpler upgrade path. I would also make it optional, so we can let the interface as is and rethink about changing it in 3.0 (no upgrade path possible for interfaces)